### PR TITLE
Allow creating both mgmt/svc cluster idempotently

### PR DESCRIPTION
**Pull request by mjlshen**
### What this PR does
**Before this PR:**
* Unable to create a mgmt and a svc cluster due to naming conflicts
* Unable to idempotently `make mgmt-cluster`/`make svc-cluster`

**After this PR:**
* Able to create a mgmt and svc cluster
* Able to idempotently `make mgmt-cluster`/`make svc-cluster`, making iterating on changes easier.

Applying the mgmt-cluster/svc-cluster targets again now results in:

```
Resource changes: 4 to modify, 10 no change.
```

Jira: [ARO-6158](https://issues.redhat.com/browse/ARO-6158)

### CODE REVIEW

File: dev-infrastructure/Makefile [hunk 1]
```diff
@@ -4,9 +4,9 @@ HCPDEVSUBSCRIPTION="ARO Hosted Control Planes (EA Subscription 1)"
 HCPDEVSUBSCRIPTIONID=1d3378d3-5a3f-4712-85a1-2485495dfc4b
 
 CURRENTUSER=$(shell az ad signed-in-user show | jq -r '.id')
-DEPLOYMENTNAME=hcp-$(USER)-dev-infra
 LOCATION?=eastus
 RESOURCEGROUP=aro-hcp-${AKSCONFIG}-$(USER)
+DEPLOYMENTNAME=$(RESOURCEGROUP)
```
Comment (mjlshen) [inline dev-infrastructure/Makefile:9]: `DEPLOYMENTNAME` needed to be unique between the mgmt/svc cluster

File: dev-infrastructure/modules/aks-cluster-base.bicep [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Constants
-param aksClusterName string = 'aro-hcp-cluster-001'
+param aksClusterName string = take('aro-hcp-${clusterType}-${uniqueString(clusterType)}', 63)
```
Comment (mjlshen) [inline dev-infrastructure/modules/aks-cluster-base.bicep:2]: This allows for a unique cluster name between mgmt/svc clusters

File: dev-infrastructure/modules/aks-cluster-base.bicep [hunk 3]
```diff
@@ -125,7 +138,7 @@ resource aks_keyvault_crypto_user 'Microsoft.Authorization/roleAssignments@2022-
   }
 }
 
-resource vnet 'Microsoft.Network/virtualNetworks@2023-09-01' = {
+resource vnet 'Microsoft.Network/virtualNetworks@2023-11-01' = {
```
Comment (mjlshen) [inline dev-infrastructure/modules/aks-cluster-base.bicep:141]: This allows vnet/subnet to behave idempotently now from https://github.com/Azure/azure-quickstart-templates/issues/2786

File: dev-infrastructure/modules/aks-cluster-base.bicep [hunk 4]
```diff
@@ -112,6 +113,18 @@ resource aks_etcd_kms 'Microsoft.KeyVault/vaults/keys@2023-07-01' = {
...
+        {
+          action: {
+            type: 'notify'
+          }
+          trigger: {
+            timeBeforeExpiry: 'P30D'
+          }
+        }
+      ]
+    }
```
Comment (mjlshen) [inline dev-infrastructure/modules/aks-cluster-base.bicep:127]: Default values that are being imported into the module

**Review by bennerv — APPROVED** [review]

**Pull request merged**